odl/tomo/geometry/geometry.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Geometry base class."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from abc import ABCMeta, abstractmethod, abstractproperty
+from future import standard_library
+from future.utils import with_metaclass
+standard_library.install_aliases()
+from builtins import object
+
+# External
+import numpy as np
+
+# Internal
+from odl.set.domain import IntervalProd
+from odl.discr.grid import RegularGrid, TensorGrid
+
+
+__all__ = ('Geometry',)
+
+
+class Geometry(with_metaclass(ABCMeta, object)):
+
+    """Abstract geometry class.
+
+    A geometry is described by
+
+    * a detector,
+    * a set of detector motion parameters,
+    * a function mapping motion parameters to the location of a
+      reference point (e.g. the center of the detector surface),
+    * a rotation applied to the detector surface, depending on the
+      motion parameters,
+    * a mapping from the motion and surface parameters to the detector
+      pixel direction to the source,
+    * optionally a mapping from the motion parameters to the source
+      position
+    """
+
+    @abstractproperty
```

Comment:
Ok. I already thought that it is related to the source but then got confused by `det_to_src`  being abstract. I'm going to remove abstract method from it.